### PR TITLE
Ran unifdef to remove INACTIVITY_TIMEOUT code blocks

### PR DESCRIPTION
**Pull request by masaori335**

### CODE REVIEW

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/293/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/279/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/275/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1974/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1866/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/406/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/538/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/301/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/287/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/283/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/413/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1874/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1982/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/546/

**Comment by oknet** [comment]
@zwoop  Do we really want remove the codes ? 

Inactivity Timeout and Active timeout have 2 types of implement in ATS.

1. by schedule_in
  - create too many event in EventQueue
2. by InactivityCop
  - makes Inactivity Timeout and Active timeout no different.

But I'm prefer InactivityCop.

+1 for the PR.

**Comment by zwoop** [comment]
@oknet The scheduling path was a long time ago deemed too expensive to handle, but honestly, not sure if that's true any more. It's possible if @SolidWallOfCode and @PSUdaemon improves on the event system, we could do the scheduling instead? The issue being the excessive adding / deleting of events (I think).

**Comment by oknet** [comment]
@zwoop 
With scheduling path, these 2 types of timeout event is put into EventQueue that is a priority queue. 
With InactivityCop, every second we copy NetVCs from open_list to cop_list and walk through cop_list that means walk through all the NetVCs twice every second.

**Comment by zwoop** [comment]
If someone wants to test / investigate the scheduled inactivity cop, I'd be fine with leaving this code as is. Also, remember that it's possible to tweak how often we run inactivity cop, we added a configuration for this (proxy.config.net.inactivity_check_frequency). Maybe we should bump the default for this to say 10s?

Alternatively, it could maybe be nice to take all the VCs that needs to be checked, and split them evenly out over (proxy.config.net.inactivity_check_frequency), such that you check

```
                    <n>
             ------------
proxy.config.net.inactivity_check_frequency
```

VCs every second?

**Comment by zwoop** [comment]
I'm making this as WIP for now, since it's not clear that we want to land this.

**Comment by bryancall** [comment]
@oknet  Have you tried running using the event handler for inactivity timeouts?

**Comment by oknet** [comment]
@zwoop According to the design of InactivityCop, the proxy.config.net.inactivity_check_frequency should less than or equal 1s otherwise the TIMEOUT event would be delay to SM.

@bryancall No, I have not tried it. But it is triggered by priority queue that is more efficient.

Currently, I consider the NetHandler is a NThread(Net Thread) inside of EThread(Event Thread).

NetHandler has its Protect Queue (enabled_list and ready_list)
EThread has its Protect Queue (ExternalQueue and ExternalQueue.local)

NetHandler has its Internal Queue (epoll fdset and ready_list)
EThread has its Protect Queue (EventQueue that is a priority queue)

And NetHandler has other queues: open_list, cop_list.

The NetVC has a dual identity:

- SM, it could be call back by EThread and others
- Event, NetHandler call back the VIO event by NetVC

An Event can not rescheduled to another EThread and also a NetVC can not rescheduled to another NetHandler since the NetVC is an 'Event' to callback SM.

An SM could be call back from any EThread and a NetVC also could be call back from any EThread before it is managed by NetHandler. Because NetHandler gives it the second identity (Event).

Therefore amc and susan designed and implement the VC migration functions.

The InactivityCop is a sub-part of NetHandler and it is not a full-featured alternative for active timeout and Inactive timeout.

We should do more for InactivityCop, 

1. A new queue to replace open_list
- to save new NetVC that did not set the timeout
- to save NetVC that has read and write disabled
- set inactivity_timeout_in to default (86400)
- check eveny 10mins or the active_queue / keepalive_queue full.
2. A new priority queue to replace cop_list
- to save any NetVC that has timeout set
- using MIN(active_timeout_in, inactive_timeout_in) as key
3. schedule InactivityCop in Negative Queue
- check cop_list like EventQueue in EThread::execute()
4. period schedule InactivityCop in EventQueue
- check open_list eveny 10mins (configureable)

**Comment by zwoop** [comment]
Hmmm, what exactly does this mean:

```
@zwoop According to the design of InactivityCop, the proxy.config.net.inactivity_check_frequency should less than or equal 1s otherwise the TIMEOUT event would be delay to SM.
```

We do

thread->schedule_every(inactivityCop, HRTIME_SECONDS(cop_freq));


Are you saying it'd be bad for the ET_NET if that schedule is > 1s? If so, we really ought to modify RecordsConfig.cc / code such that it can not be configured differently. Also, it's then really odd that this is in seconds, because the only two viable options would be 0 or 1, right ? And I can't imagine 0 is a good value. The conclusion would be that if "1" is the only viable option, we should either

1) Eliminate this configuration entirely.

2) Change it to be in milli-seconds, with a default (and max) of 1000.

**Comment by oknet** [comment]
@zwoop 
We set timeout with set_inactivity_timeout(HRTIME_SECONDS(x)) that means ATS call back TIMEOUT event between x th and x+1 th second.  
To achieve this, ATS must check timeout every second or less otherwise the TIMEOUT event will be send between (x+cop_freq) th and (x+cop_freq+1) th second at latest.

**Comment by bryancall** [comment]
[approve ci]

**Comment by zwoop** [comment]
@oknet What should we do here? Do this (merge) or keep it ?

**Comment by oknet** [comment]
@zwoop +1 merge since the `event mode` may not working with HttpSM now.

To operation on the timeout event, the HttpSM should acquire the lock of vc->nh or the vc->nh has been locked. I don't think the HttpSM does it as well.

File: iocore/net/UnixNetVConnection.cc [hunk 1]
```diff
@@ -98,19 +85,8 @@ close_UnixNetVConnection(UnixNetVConnection *vc, EThread *t)
...
-  if (vc->active_timeout) {
-    vc->active_timeout->cancel_action(vc);
-    vc->active_timeout = nullptr;
-  }
-#else
   vc->next_inactivity_timeout_at = 0;
   vc->next_activity_timeout_at   = 0;
-#endif
 
   vc->inactivity_timeout_in = 0;
```
Comment (oknet) [inline iocore/net/UnixNetVConnection.cc]: clang format here.

Comment (masaori335) [inline iocore/net/UnixNetVConnection.cc]: clang-format is already applied:)

**Comment by masaori335** [comment]
[approve ci clang-analyzer]

**Comment by masaori335** [comment]
Rebased and fixed conflicts. Please take another look.

**Pull request merged**